### PR TITLE
q3map2: import minimap stuff from NetRadiant

### PR DESCRIPTION
**Pull request by illwieckz**
Beware, this PR is large, I tried to keep as most as history possible.

I had a bad surprise while doing it since I discovered this minimap code was relying on two non-minimap code that were commited before so I had to add them too… (you will find these non-minimap change in the two older commits of this PR)

GitHub is annoying to not offer an option to ignore whitespace while rendering diff…

It's easy to test:

```sh
./install/q3map2 -minimap ~/.q3a/baseq3/maps/q3dm1sample.bsp
```

Please do not merge too quickly this PR, it's not bad to look at it again after some days has passed… :-)

### CODE REVIEW

**Comment by illwieckz** [comment]
hmm, if q3map2 produces minimap well, it now segfault on bsp compilation… so it's not ready yet. ^_^

**Comment by illwieckz** [comment]
OK I fixed the segfault on bsp compilation, will experiment more later.

File: tools/quake3/q3map2/bsp.c [hunk 1]
```diff
@@ -842,11 +841,30 @@ int BSPMain( int argc, char **argv ){
...
+		{
 			Sys_Printf( "-bsp argument unnecessary\n" );
 		}
-		else{
-			Sys_FPrintf( SYS_WRN, "WARNING: Unknown option \"%s\"\n", argv[ i ] );
+		else if( !strcmp( argv[ i ], "-deep" ) )
+		{
+			Sys_Printf( "Deep BSP tree generation enabled\n" );
+			deepBSP = qtrue;
+		}
```
Comment (TTimo) [inline tools/quake3/q3map2/bsp.c]: duplicate code

File: tools/quake3/q3map2/bsp.c [hunk 2]
```diff
@@ -842,11 +841,30 @@ int BSPMain( int argc, char **argv ){
 			Sys_Printf( "Debug portal surfaces enabled\n" );
 			debugPortals = qtrue;
 		}
-		else if ( !strcmp( argv[ i ], "-bsp" ) ) {
+		else if ( !strcmp( argv[ i ], "-altsplit" ) )
+		{
+			Sys_Printf( "Alternate BSP splitting (by 27) enabled\n" );
+			bspAlternateSplitWeights = qtrue;
+		}
```
Comment (TTimo) [inline tools/quake3/q3map2/bsp.c]: this doesn't do anything in this commit, maybe remove?

File: tools/quake3/q3map2/bsp.c [hunk 3]
```diff
@@ -842,11 +841,30 @@ int BSPMain( int argc, char **argv ){
...
+		else if ( !strcmp( argv[ i ], "-deep" ) )
+		{
+			Sys_Printf( "Deep BSP tree generation enabled\n" );
+			deepBSP = qtrue;
+		}
```
Comment (TTimo) [inline tools/quake3/q3map2/bsp.c]: Do a more complete pass in netradiant q3map2 to make sure everything related to deepBSP is caught, or remove as well?

**Comment by TTimo** [comment]
See comments. A bit concerned about this bringing in some changes to the BSP splitting code.

File: tools/quake3/q3map2/bsp.c (hunk 3 above)
Comment (illwieckz) [reply tools/quake3/q3map2/bsp.c]: I re-read each commit entirely side by side and yes some stuff was missing, what was missing is now imported too.

**Pull request merged**